src/transformers/generation_beam_constraints.py

```diff
@@ -249,26 +387,33 @@ def advance(self):
         Though we don't care which constraint is fulfilled first, if we are in the progress of fulfilling a constraint,
         that's the only one we'll return.
         """
+        token_list = []
         if self.inprogress_constraint is None:
-            token_list = []
             for constraint in self.pending_constraints:  # "pending" == "unfulfilled yet"
                 advance = constraint.advance()
-                token_list.append(advance)
+                if isinstance(advance, int):
+                    token_list.append(advance)
+                elif isinstance(advance, list):
+                    token_list = token_list + advance
```

Comment:
Really picky NIT: `token_list.extend(advance)` is more pythonistic IMO.
It's also faster since it doesn't create a new  list.
It makes the understanding of what goes on easier IMO, because `.extend` only exists of iterable types, whereas without context `token_list` and `advance` could be read as `int` by an external reader
It also matches the `.append`  above.

Feel free to ignore

Reply:
agreed!